tests/unit/manage/views/test_organizations.py

```diff
@@ -1590,11 +1590,16 @@ def test_add_organization_project_new_project(
         view = org_views.ManageOrganizationProjectsViews(organization, db_request)
         result = view.add_organization_project()
 
-        # The project was created, and belongs to the organization.
+        # The project was created
         project = (
             db_request.db.query(Project).filter_by(name="fakepackage").one_or_none()
         )
         assert project is not None
+
+        # Refresh the project in the DB session to ensure it is not stale
+        db_request.db.refresh(project)
```

Comment:
What I don't fully understand is how this is possible, if the line above this is the first time we're querying for this object from the DB.

Is it possible that these lines are somehow causing this?
https://github.com/pypi/warehouse/blob/6e3fb21a454f26de9205817ea976beb7cdf1e51d/warehouse/organizations/services.py#L520-L523

Reply:
These lines have no effect on the test here, and neither does additional flushes elsewhere. I don't have an answer for why this is happening in this test 🤷